src/main/scala/org/apache/spark/sql/pulsar/PulsarSource.scala

```
@@ -63,10 +79,79 @@ private[pulsar] class PulsarSource(
     initialTopicOffsets
```

We need to throw an exception from getOffset to make sure it is not getting called.

---

Done